src/main/java/school/lemon/changerequest/java/fractionnumber/FractionNumber.java

```diff
@@ -0,0 +1,110 @@
+package school.lemon.changerequest.java.fractionnumber;
+
+public class FractionNumber {
+    private int dividend;
+    private int divisor;
+
+    public static FractionNumber ONE = new FractionNumber(1, 1);
+    public static FractionNumber ZERO = new FractionNumber(0, 1);
+
+    private void normalize() {
+        if (dividend == 0) {
+            divisor = 1;
+            return;
+        }
+        int greatestCommonDivisor = greatestCommonDivisor(dividend, divisor);
+        dividend /= greatestCommonDivisor;
+        divisor /= greatestCommonDivisor;
+        if (divisor < 0) {
+            dividend *= -1;
+            divisor *= -1;
+        }
+    }
+
+    private int greatestCommonDivisor(int a, int b) {
+        while (b != 0) {
+            a %= b;
+            int tmp = a;
+            a = b;
+            b = tmp;
+        }
+        return a;
+    }
+
+    public FractionNumber(int dividend, int divisor) {
+        if (divisor == 0)
+            throw new IllegalArgumentException("Divisor is equal to zero");
+        this.divisor = divisor;
+        this.dividend = dividend;
+        normalize();
+    }
+
+    public FractionNumber(int dividend) {
```

Comment:
What if I'll change the task, so that the default `divisor` value would be `3`? Will it automatically normalize the value?

I'd recommend re-using of the existing constructors when it is possible — just class `this(dividend, 1)` and you'll automatically use existing logic.